crates/provider-sdk/src/provider.rs

```diff
@@ -90,81 +692,76 @@ pub struct ProviderConnection {
     links: Arc<RwLock<HashMap<SourceId, InterfaceLinkDefinition>>>,
 
     /// NATS client used for performing RPCs
-    rpc_client: RpcClient,
+    nats: Arc<async_nats::Client>,
 
     /// Lattice name
     lattice: String,
-
-    /// Data received from the host at startup
-    host_data: Arc<HostData>,
-
-    /// Handles for every NATS listener that is created and used by the Provider, kept
-    /// around so they can appropriately be `Drop`ed when this `ProviderConnection` is
-    _listener_handles: Arc<Mutex<Vec<tokio::task::JoinHandle<Result<()>>>>>,
+    host_id: String,
+    link_name: String,
+    provider_key: String,
 
     /// Mapping of NATS subjects to dynamic function information for incoming invocations
     #[allow(unused)]
     incoming_invocation_fn_map: Arc<WrpcInvocationLookup>,
 }
 
-impl std::fmt::Debug for ProviderConnection {
-    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+impl fmt::Debug for ProviderConnection {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
         f.debug_struct("ProviderConnection")
-            .field("provider_id", &self.host_data.provider_key)
-            .field("host_id", &self.host_data.host_id)
-            .field("link", &self.host_data.link_name)
+            .field("provider_id", &self.provider_key())
+            .field("host_id", &self.host_id)
+            .field("link", &self.link_name)
             .field("lattice", &self.lattice)
             .finish()
     }
 }
 
+/// Returns a provider-specific [`wrpc_transport::Client`]
+pub fn wrpc_client(
+    nats: Arc<async_nats::Client>,
+    lattice: &str,
+    provider_key: &str,
+    target: &str,
+) -> wasmcloud_core::wrpc::Client {
+    let mut headers = HeaderMap::new();
+    headers.insert("source-id", provider_key);
+    headers.insert("target-id", target);
+    wasmcloud_core::wrpc::Client::new(nats, lattice, target, headers)
+}
```

Comment:
NIT: Should this be public?  maybe `pub(crate)`?

The vast majority of the time, people should be getting their client from an active connection via the SDK, not building their own, since you generally don't the `provider_key` without reading host data, and that information goes into the `ProviderConnection`.